### PR TITLE
Eth USD price

### PR DESCRIPTION
**Pull request by prxt6529**

### CODE REVIEW

File: src/ethPriceLoop/eth_usd_price.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+interface CurrentResponse {
+  data: {
+    price: number;
+  };
+}
+
+const logger = Logger.get('ETH_PRICE');
+
+export async function getNewEthUsdPrice(reset: boolean) {
```
Comment (GelatoGenesis) [inline src/ethPriceLoop/eth_usd_price.ts:27]: Using get and fetch prefixes in modifying functions. Still not a fan of that :)

```
getNewUsdPrice -> synchroniseEthUsdPrice
fetchHistoricData -> synchroniseHistoricEthUsdPriceData
fetchCurrentData -> synchroniseLatestEthUsdPriceData
```

File: src/ethPriceLoop/package.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    "sls-deploy:staging": "node ../../node_modules/serverless/bin/serverless.js deploy --stage=staging --region=eu-west-1"
+  },
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "@types/aws-lambda": "^8.10.110",
+    "esbuild": "^0.17.5"
+  },
+  "dependencies": {
+    "alchemy-sdk": "^2.4.1"
```
Comment (GelatoGenesis) [inline src/ethPriceLoop/package.json:20]: Why do you need this here?

File: src/ethPriceLoop/db.eth_price.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    `
+      SELECT * from ${ETH_PRICE_TABLE}
+      WHERE timestamp_ms <= ?
+      ORDER BY timestamp_ms DESC
+      LIMIT 1
+  `,
+    [timestampMs]
+  );
+
+  return price[0].usd_price ?? 0;
```
Comment (GelatoGenesis) [inline src/ethPriceLoop/db.eth_price.ts:27]: This would result to an error if there are 0 rows. `price[0]?.usd_price ?? 0` would be more correct.

Comment (prxt6529) [reply src/ethPriceLoop/db.eth_price.ts]: fixed

File: src/ethPriceLoop/package.json (hunk 2 above)
Comment (prxt6529) [reply src/ethPriceLoop/package.json]: i don't, removed

File: src/ethPriceLoop/eth_usd_price.ts (hunk 1 above)
Comment (prxt6529) [reply src/ethPriceLoop/eth_usd_price.ts]: updated - used 'sync' to have a bit shorter name

**Review by GelatoGenesis — APPROVED** [review]

**Pull request merged**